metis-pattern-analysis/src/main/java/eu/europeana/patternanalysis/ProblemPatternAnalyzer.java

```diff
@@ -53,51 +71,176 @@ public List<ProblemPattern> analyzeRecord(String rdfString) throws Serialization
    */
   public List<ProblemPattern> analyzeRecord(RDF rdf) {
     final List<ProxyType> providerProxies = getProviderProxies(rdf);
-    final List<String> titles = providerProxies.stream().map(EuropeanaType::getChoiceList).flatMap(Collection::stream)
-                                               .filter(Choice::ifTitle).map(Choice::getTitle)
-                                               .map(Title::getString)
-                                               .filter(StringUtils::isNotBlank)
-                                               .map(String::trim)
-                                               .collect(Collectors.toList());
-    final List<String> descriptions = providerProxies.stream().map(EuropeanaType::getChoiceList).flatMap(Collection::stream)
-                                                     .filter(Choice::ifDescription).map(Choice::getDescription)
-                                                     .map(Description::getString)
-                                                     .filter(StringUtils::isNotBlank)
-                                                     .map(String::trim)
-                                                     .collect(Collectors.toList());
+    final List<Choice> choices = providerProxies.stream().map(EuropeanaType::getChoiceList).flatMap(Collection::stream)
+                                                .collect(toList());
+
+    final List<String> titles = getChoicesInStringList(choices, Choice::ifTitle, Choice::getTitle, LiteralType::getString);
+    final List<String> descriptions = getChoicesInStringList(choices, Choice::ifDescription, Choice::getDescription,
+        ResourceOrLiteralType::getString);
+    final List<String> identifiers = getChoicesInStringList(choices, Choice::ifIdentifier, Choice::getIdentifier,
+        LiteralType::getString);
     final String rdfAbout = rdf.getProvidedCHOList().stream().filter(Objects::nonNull).findFirst()
                                .map(ProvidedCHOType::getAbout).orElse(null);
-    return computeProblemPatterns(rdfAbout, titles, descriptions);
+    return computeProblemPatterns(rdfAbout, titles, descriptions, identifiers);
+  }
+
+  private <T> List<String> getChoicesInStringList(List<Choice> choices, Predicate<Choice> choicePredicate,
+      Function<Choice, T> choiceGetter, Function<T, String> getString) {
+    return choices.stream().filter(choicePredicate).map(choiceGetter).map(getString).collect(toList());
```

Comment:
does it worth to keep or remove ```null``` items? it will affect all checks.

Reply:
if we leave ```nulls``` that means the field is missing? unit test for this case.

Reply:
Not completely sure this could happen in Jibx deserialization(mostly because of the schema) but doesn't hurt to filter those.
Not sure I can make it "happen" in the unit test with xml but I will try.

Reply:
I've added some more test but I could not represent a null value on deserialization for a Choice.
I could though for a the list of Choices so I've added that.

Reply:
:+1: